src/mpid/ch4/netmod/ofi/ofi_rma.h

```diff
@@ -390,10 +390,15 @@ static inline int MPIDI_OFI_do_put(const void *origin_addr,
 
     MPIDI_Datatype_check_contig_size_lb(target_datatype, target_count, target_contig, target_bytes,
                                         target_true_lb);
-    MPIDI_Datatype_check_contig_size_lb(origin_datatype, origin_count, origin_contig, origin_bytes,
-                                        origin_true_lb);
-
-    MPIR_ERR_CHKANDJUMP((origin_bytes != target_bytes), mpi_errno, MPI_ERR_SIZE, "**rmasize");
+    if (target_datatype == origin_datatype && target_count == origin_count) {
+        origin_contig = target_contig;
+        origin_bytes = target_bytes;
+        origin_true_lb = target_true_lb;
+    } else {
+        MPIDI_Datatype_check_contig_size_lb(origin_datatype, origin_count, origin_contig,
+                                            origin_bytes, origin_true_lb);
+        MPIR_ERR_CHKANDJUMP((origin_bytes != target_bytes), mpi_errno, MPI_ERR_SIZE, "**rmasize");
+    }
 
```

Comment:
Can we define a wrapper function for this optimization as the same pattern is used in several places

Reply:
Done. Could you please take another look?